src/components/ListItem.jsx

```diff
@@ -1,5 +1,59 @@
 import './ListItem.css';
+import { useState, useEffect } from 'react';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listToken }) {
+	const one_day_in_ms = 24 * 60 * 60 * 1000;
+	// const one_day_in_ms = 60 * 2 * 1000; // 120 seconds for testing the reset timeframe
+	const currentDate = new Date();
+	const currentTimeInMilliseconds = Math.floor(currentDate.getTime());
+	const dateLastPurchasedInMilliseconds =
+		item && item.dateLastPurchased
+			? item.dateLastPurchased.seconds * 1000
+			: null;
+	const timeElapsed =
+		currentTimeInMilliseconds - dateLastPurchasedInMilliseconds;
+
+	const [isPurchased, setIsPurchased] = useState(item.isChecked);
```

Comment:
When I attempt to search/filter the list, this item.isChecked throws a typeError. This should be fixed before merge

![react-typeError](https://user-images.githubusercontent.com/89806097/181920111-ed5a79b9-5e89-43a3-9e6f-68c8575e94a6.png)

Reply:
Hey @lindseyindev I'm having a hard time figuring this one out. Each error I fix leads to a new error. It seems like things are breaking during the search filter because there is no item. I added some ternary checks in the input fields that check if there is an item or not. However, now I am running into an issue where the updateItem function is not working because there is no itemData. And I'm not quite sure how to resolve this. I can either push the code I have now or maybe take some screenshots? I'm not sure that what I've done with the ternary operators is a good idea or best practice.

Reply:
I see this now.. honestly I thought it would be an easy update to remove isChecked and initialize state with false but the item.id is failing now too. As I'm developing I also removed the updateItem call in the useEffect so I don't have to worry about creating an infinite loop and would recommend that too if you do any debugging. I am just testing so far with console logs and won't add the firebase call back until I'm sure that this works in the browser first.

Reply:
Hello! I have a suggestion for this one - I noticed that when we were mapping through searchResults in `List`, we didn't pass item in as a prop into `<ListItem />`. There are two of these components because we have a ternary with two maps, so we need to pass item as a prop into the second `<ListItem />` as well. When I passed it down, it was able to search on my end!

Reply:
It looks like in List.jsx, the props passed to `<ListItem />` are not always the same.

```jsx
{!searchQuery
	? data.map((item) => (
			<ListItem key={item.id} item={item} listToken={listToken} />
			))
	: searchResults.map((item) => (
			<ListItem key={item.id} name={item.name} listToken={listToken} />
			))}
```

There is no `name` prop. It should be an `item` prop instead when there is a search query, which explains why you get the error about `item.isChecked`.

Reply:
Good catch @mira-kine!

Reply:
@mira-kine great catch!! I think we had changed this prop from name, to item, so we'd have access to all the properties of the item object, not just the name. We obviously forgot to make that same change on the search results! I think this fixed the issues with filtering!